### DISCUSSION TITLE
Where is firmware micropython for ESP32-H2?

### DISCUSSION THREAD

**Discussion by Stefa23Lan** [micropython/micropython#13201] [ESP32] [answered]
Where is firmware micropython for ESP32-H2?

**Comment by robert-hh** [comment]
There is none, AFAIK. 

Edit: The core features may work with a ESP32C3 firmware, but no Zigbee, and attempts to use WiFi may cause unpredictable results.

**Comment by Stefa23Lan** [comment]
Thk, for your answer. Not wifi on esp32-h2.
Is there a version planned? if yes, when?

**Comment by Ikif** [comment]
Micropython goes to support ESP32-H2 ?

**Comment by mattytrentini** [comment, marked as answer]
I don't think any of the core team _have_ an ESP32-H2 yet (I don't either). However, a lot of the heavy lifting ought to be able to leverage the effort invested in the ESP32-C6 support, [currently in a PR](https://github.com/micropython/micropython/pull/11869). The H2 is supported by [roughly the same IDF versions](https://github.com/espressif/esp-idf?tab=readme-ov-file#esp-idf-release-and-soc-compatibility).

Of course, if someone wants to raise a PR I'm sure it'll be considered!

**Comment by shariltumin** [comment]
If you're interested, please take a look at:

https://github.com/shariltumin/esp32-h2-w5500-micropython
